### DISCUSSION TITLE
align numeric fields at decimal point

### DISCUSSION THREAD

**Discussion by NickCrews** [directus/directus#24937] [Feature Requests]
### Describe the Improvement

Currently, numeric columns can be aligned left, center, or right, just like all the rest. This makes it hard to compare the relative size of things at-a-glance. I want larger numbers to stick further left than smaller numbers. Right-align works if all numbers have the same number of digits to the right of the decimal point, but if there is a mix then it doesn't work.
<img width="360" alt="image" src="https://github.com/directus/directus/assets/10820686/5871860d-bff4-4f44-947d-676159b020c1">

This is a small UX thing, not a big deal, but if it's not too hard then that would be great. If it's a pain then feel free to just close this out. If you point me in the right direction then I can take a stab at a PR, though I am a noob with typescript (though am an experienced python engineer).

Other more generic solution would be if we added a more general formatting API to fields (I bet there is already an issue about that) so that this is the user's responsibility/power.

Thank you!

**Comment by azrikahar** [comment]
#19467 did add a similar QoL change to metric panels with the addition of the [`formatNumber()`](https://github.com/directus/directus/blob/25190b42f1dea8ace49d0ae92daaed454b7d884c/app/src/utils/format-number.ts#L133) util which uses [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat), with the following options:

![](https://github.com/directus/directus/assets/42867097/30ccb073-f8fb-46a0-9a09-f127c7c76f58)

which will then be passed over here to be used in the metric panel:

https://github.com/directus/directus/blob/25190b42f1dea8ace49d0ae92daaed454b7d884c/app/src/panels/metric/panel-metric.vue#L148-L155

By doing a very quick and rough test (an actual PR will likely need to be more elaborate) and changing the following:

https://github.com/directus/directus/blob/25190b42f1dea8ace49d0ae92daaed454b7d884c/app/src/displays/formatted-value/formatted-value.vue#L110

to something such as:

```js
value = formatNumber(parseFloat(value), locale.value, { minimumFractionDigits: 2 });
```

It does seem to work:

![](https://github.com/directus/directus/assets/42867097/91622a4c-da02-4a33-bf00-93dbb1bd63ae)

That being said, we do need be careful not to inundate users with _too_ many options (understandably this is balancing thing and debatable!). Looking back, it also seems like the great update to formatted-value display via #9773 did actually have the `minimumFractionDigits` option, but was then simplified to the `n()` function as mentioned in https://github.com/directus/directus/pull/9773#discussion_r754590543.

I would suggest creating a [custom display](https://docs.directus.io/extensions/displays.html) which uses `Intl.NumberFormat` if you need such flexibility right away, but a PR would be welcomed as I do personally agree such UX improvement (especially in the case of currency) should be a core functionality.

**Comment by LevelUpGit** [comment]
We wonder also why we can't show decimals by default, it would be small but good improvement.
